src/processors/JsonSchemaInputProcessor.ts

```diff
@@ -49,144 +53,155 @@ export class JsonSchemaInputProcessor extends AbstractInputProcessor {
     const deRefOption: $RefParser.Options = {
       continueOnError: true,
       dereference: { circular: true },
     };
     Logger.debug(`Trying to dereference all $ref instances from input, using option ${JSON.stringify(deRefOption)}.`);
     try {
       await refParser.dereference(localPath, input, deRefOption);
-    } catch (e) {
+    } catch (e: any) {
       const errorMessage = `Could not dereference $ref in input, is all the references correct? ${e.message}`;
       Logger.error(errorMessage, e);
       throw new Error(errorMessage);
     }
-    const parsedSchema = Schema.toSchema(input);
-    Logger.debug('Successfully dereferenced all $ref instances from input.', parsedSchema);
-    commonInputModel.models = JsonSchemaInputProcessor.convertSchemaToCommonModel(parsedSchema);
-    return commonInputModel;
+    Logger.debug('Successfully dereferenced all $ref instances from input.', input);
   }
 
   /**
+   * Each schema must have a name, so when later interpreted, the model have the most accurate model name.
+   * 
    * Reflect name from given schema and save it to `x-modelgen-inferred-name` extension.
    * 
+   * This reflects all the common keywords that are shared between draft-4, draft-7 and Swagger 2.0 Schema
+   * 
    * @param schema to process
    * @param namesStack is a aggegator of previous used names
    * @param name to infer
    * @param isRoot indicates if performed schema is a root schema
    */
   // eslint-disable-next-line sonarjs/cognitive-complexity
   static reflectSchemaNames(
-    schema: Schema | boolean,
-    namesStack: Record<string, number> = {}, 
+    schema: Draft7Schema | boolean,
+    namesStack: Record<string, number>,
     name?: string,
     isRoot?: boolean,
-  ): Schema | boolean {
+  ): any {
     if (typeof schema === 'boolean') {return schema;}
 
     schema = Object.assign({}, schema);
     if (isRoot) {
       namesStack[String(name)] = 0;
-      schema[this.MODELGEN_INFFERED_NAME] = name;
+      (schema as any)[this.MODELGEN_INFFERED_NAME] = name;
       name = '';
-    } else if (name && !schema[this.MODELGEN_INFFERED_NAME]) {
+    } else if (name && !(schema as any)[this.MODELGEN_INFFERED_NAME]) {
       let occurrence = namesStack[String(name)];
       if (occurrence === undefined) {
         namesStack[String(name)] = 0;
       } else {
         occurrence++;
       }
       const inferredName = occurrence ? `${name}_${occurrence}` : name;
-      schema[this.MODELGEN_INFFERED_NAME] = inferredName;
+      (schema as any)[this.MODELGEN_INFFERED_NAME] = inferredName;
     }
 
     if (schema.allOf !== undefined) {
-      schema.allOf = schema.allOf.map((item, idx) => this.reflectSchemaNames(item, namesStack, this.ensureNamePattern(name, 'allOf', idx)));
+      schema.allOf = (schema.allOf as any[]).map((item: any, idx: number) => JsonSchemaInputProcessor.reflectSchemaNames(item, namesStack, this.ensureNamePattern(name, 'allOf', idx)));
```

Comment:
You can use `this` keyword for static methods/properties in static methods/properties, so why you changed it to class reference?

Reply:
Thaaats a good question, I think it was because I tried different restructures first and then this happened 😄 Changed!